test/utils/ui.test.helper.ts

```diff
@@ -53,18 +53,30 @@ export const returnElementText = async (
   const returnedText = await page.$eval(selector, element =>
     element.textContent?.trim()
   );
-
   return returnedText;
 };
 
 /**
- * Returns boolean result if element existence
+ * Awaits element to exist on page
  * @param selector of element
  */
 export const verifyElementExistOnPage = async (
   page: puppeteer.Page,
   selector: string
-) => !!(await page.$(selector));
+) => await page.waitForSelector(selector, { hidden: false, visible: true });
+
+export const loading = async (page: puppeteer.Page) => {
+  let progressbarElement = await page.$('[role="progressbar"]');
+  let invisible = await page.waitForSelector('[role="progressbar"]', {
+    hidden: true,
+  });
+
+  if (progressbarElement) {
+    invisible;
```

Comment:
what is that supposed to do?

Reply:
Waits for loading indicator to display and hide.

Reply:
I meant the expression on line 75.

Reply:
Addressed